src/Query/QueryBuilder.php

```diff
@@ -203,6 +203,14 @@ public function expr(): ExpressionBuilder
         return $this->connection->createExpressionBuilder();
     }
 
+    /**
+     * Returns a fresh query builder instance that can be used to build a subquery.
+     */
+    public function sub(): self
```

Comment:
Although it can be used to build a sub query, it does not have to, and the two instances maintain no link to each other. Did you consider other names for this method? From what I can tell , the important thing is that both QBs share the same connection. But maybe the intent expressed in the comment is more important than the implementation.

Reply:
I chose the name because creating a subquery is the usual use-case for calling that method. Yes, you can "abuse" it for other things, but why should I reflect this in the name?

That being said, we can change the name before the 4.4.x release. If you have a better idea, feel free to drop it here or open a PR. 🙂